File: benchmark/lua/bsearch.lua
local arr = {}
for i = 1, 100000 do
    arr[i] = (i - 1) * 2
end
local start = os.clock()
local found = 0
for j = 1, 10000 do
    local target = (j - 1) * 20
    local left = 1
    local right = 100000
    while left <= right do
        local mid = math.floor((left + right) / 2)
        if arr[mid] == target then
            found = found + 1
            break
        end
        if arr[mid] < target then
            left = mid + 1
        else
            right = mid - 1
        end
    end
end
local finish = os.clock()
print("Found: " .. found)
print(string.format("Time: %.0fms", (finish - start) * 1000))

File: benchmark/python/bsearch.py
import time

arr = [i * 2 for i in range(100000)]
start = time.time() * 1000
found = 0
for j in range(10000):
    target = j * 20
    left = 0
    right = 99999
    while left <= right:
        mid = (left + right) // 2
        if arr[mid] == target:
            found += 1
            break
        if arr[mid] < target:
            left = mid + 1
        else:
            right = mid - 1
end = time.time() * 1000
print(f"Found: {found}")
print(f"Time: {end - start:.0f}ms")

